pkg/mount/magic.go

```diff
@@ -0,0 +1,212 @@
+	HUGETLBFS = []byte{0x95, 0x84, 0x58, 0xf6}
+	JFFS2     = []byte{0x72, 0xb6}
+	JFS       = []byte{0x31, 0x53, 0x46, 0x4a}
+	MTD       = []byte{0x11, 0x30, 0x78, 0x54}
+	NFS       = []byte{0x69, 0x69}
+	NILFS     = []byte{0x34, 0x34}
+	NSFS      = []byte{0x6e, 0x73, 0x66, 0x73}
+	// From docs, not tested.
+	NTFS       = []byte{0xeb, 0x52, 0x90, 'N', 'T', 'F', 'S', ' ', ' ', ' ', ' '}
+	OCFS2      = []byte{0x74, 0x61, 0x63, 0x6f}
+	OPENPROM   = []byte{0x9f, 0xa1}
+	OVERLAYFS  = []byte{0x79, 0x4c, 0x76, 0x30}
+	PIPEFS     = []byte{0x50, 0x49, 0x50, 0x45}
+	PROC       = []byte{0x9f, 0xa0}
+	PSTOREFS   = []byte{0x61, 0x65, 0x67, 0x6C}
+	QNX4       = []byte{0x00, 0x2f}
+	QNX6       = []byte{0x68, 0x19, 0x11, 0x22}
+	RAMFS      = []byte{0x85, 0x84, 0x58, 0xf6}
+	RDTGROUP   = []byte{0x76, 0x55, 0x82, 1}
+	ROMFS      = []byte{0x72, 0x75}
+	SECURITYFS = []byte{0x73, 0x63, 0x66, 0x73}
+	SELINUX    = []byte{0xf9, 0x7c, 0xff, 0x8c}
+	SMACK      = []byte{0x43, 0x41, 0x5d, 0x53}
+	SMB        = []byte{0x51, 0x7B}
+	SOCKFS     = []byte{0x53, 0x4F, 0x43, 0x4B}
+	SYSFS      = []byte{0x62, 0x65, 0x65, 0x72}
+	TMPFS      = []byte{0x01, 0x02, 0x19, 0x94}
+	TRACEFS    = []byte{0x74, 0x72, 0x61, 0x63}
+	UBIFS      = []byte{0x24, 0x05, 0x19, 0x05}
+	UDF        = []byte{0x15, 0x01, 0x33, 0x46}
+	USBDEVICE  = []byte{0x9f, 0xa2}
+	V9FS       = []byte{0x01, 0x02, 0x19, 0x97}
+	XENFS      = []byte{0xab, 0xba, 0x19, 0x74}
+	ZONEFS     = []byte{0x5a, 0x4f, 0x46, 0x53}
+	ZSMALLOC   = []byte{0x58, 0x29, 0x58, 0x29}
+)
+
+type magic struct {
+	magic []byte
+	off   int
+	name  string
+	flags uintptr
+}
+
+// magics maps the statfs fsword (an ambiguous type, to say the least) to strings
+// compatible with what is found in /proc/filesystems.
+// One file system in particular shares a single magic for several types.
+// For that reason, and reasons of space, this is a list, not a map.
+// Performance is not really an issue: it is a short list, and there are simply
+// not enough block devices/file systems for it to really matter.
+// The ordering for the identical magic number file systems matters: ext4 is more
+// desirable than ext2, so, we want to find ext4 first.
+// Also, the for loop across this will bail at the first -1, so as to save some time.
+// The order should NOT BE ALPHABETIC, therefore; it should be ordered with known systems
+// first, and, to break ties, with the most desirable of those systems first.
+var magics = []magic{
+	// From the filesystems magic:
+	// 0x438   leshort         0xEF53          Linux
+	{magic: EXT4, name: "ext4", off: 0x438},
+	{magic: EXT3, name: "ext3", off: 0x438},
+	{magic: EXT2, name: "ext2", off: 0x438},
+	// We will always mount vfat; it's backward compatible (we think?)
+	{magic: MSDOS, name: "vfat", off: 0},
+	{magic: SQUASHFS, name: "squashfs", flags: MS_RDONLY, off: 0},
+	{magic: ISOFS, name: "iso9660", flags: MS_RDONLY, off: 32768},
+	{magic: VFAT, name: "vfat", off: 0},
+	{magic: XFS, name: "xfs", off: 0},
+
+	//
```

Comment:
Can these just go into a separate unsupportedMagics list since we're not using them?

Reply:
uh, yeah, what was I thinking here?